src/folly/FollyKeepaliveTimer.cpp

```diff
@@ -27,8 +27,18 @@ void FollyKeepaliveTimer::schedule() {
   eventBase_.runAfterDelay(
       [this, running]() {
         if (*running) {
-          connection_->sendKeepalive();
-          schedule();
+          if (pending_) {
+            stop();
+
+            connection_->outputFrameOrEnqueue(
```

Comment:
Yep, will do.  I wonder whether it should be a terminateWithPrejudice method.  I'm never sure this error frame actually gets sent before termination, and its repeated inside ConnectionAutomaton also.